src/Machete/ValueExtensions.cs

```diff
@@ -174,5 +174,25 @@ public static T Or<T>(this Value<T> value, T other)
 
             return getter(source.Value) ?? Schema.Layout.Missing<T>();
         }
+
+        /// <summary>
+        /// Returns a count of segments in <see cref="ValueList{TValue}"/>
+        /// </summary>
+        /// <param name="source"></param>
+        /// <typeparam name="TValue"></typeparam>
+        /// <returns></returns>
+        public static int Count<TValue>(this ValueList<TValue> source)
```

Comment:
Because it forces the entire list to be resolved, versus one at a time. It just makes it easy to fall into the for(;;) loop mentality which can be less efficient when searching for things.